lnwire/allow_htlc_request.go

```diff
@@ -0,0 +1,107 @@
+package lnwire
+
+import (
+	"fmt"
+	"io"
+)
+
+// AllowHTLCRequest is sent to Bob from Alice. Alice wants to know if Bob is able to
+// make htlc chain with some other node with a given amount.
+// This message is sent to all nodes in possible HTLC chain to ensure that they can
+// accept it.
+// It is used to validate/filter possible paths in routing.
+type AllowHTLCRequestMessage struct {
```

Comment:
I don't think this message nor its inclusion in the routing protocol is necessary. Instead, Alice should simply attempt to send the HTLC factoring in the latest information she has related to the total and available capacity of all channels in the route. In the case that somewhere along the route, a channel has insufficient capacity, then the node will start to send `CancelHTLC` messages backwards along the route thereby failing the routing attempt. 

Using the policy described above minimizes latency and extra network traffic -- we simply fail quickly if the available flow within the path is insufficient. Also if we assume a network setting with very rapid payments, then the available channel flow which was once marked as available my be depleted by the time this message gets back to the sender.